src/cpp/subscriber/SubscriberHistory.cpp

```diff
@@ -77,300 +90,226 @@ bool SubscriberHistory::received_change(
         CacheChange_t* a_change,
         size_t unknown_missing_changes_up_to)
 {
-
     if (mp_reader == nullptr || mp_mutex == nullptr)
     {
         logError(RTPS_HISTORY, "You need to create a Reader with this History before using it");
         return false;
     }
 
     std::lock_guard<std::recursive_timed_mutex> guard(*mp_mutex);
+    return receive_fn_(a_change, unknown_missing_changes_up_to);
+}
 
-    //NO KEY HISTORY
-    if (mp_subImpl->getAttributes().topic.getTopicKind() == NO_KEY)
+bool SubscriberHistory::received_change_keep_all_no_key(
+        CacheChange_t* a_change,
+        size_t unknown_missing_changes_up_to)
+{
+    // TODO(Ricardo) Check
```

Comment:
Can this TODO comment be removed or addressed?

Reply:
This is a backport, and this TODO has not been addressed on the mainline.